src/index.js

```diff
@@ -610,11 +630,34 @@ export default class HyperScreen extends React.Component {
         return response.text();
       })
       .then((responseText) => {
-        const doc = this.parser.parseFromString(responseText);
+        let doc = this.parser.parseFromString(responseText);
+        let error = false;
         const animations = createAnimations(doc);
         const stylesheets = Stylesheets.createStylesheets(doc);
         ROUTE_KEYS[getHrefKey(url)] = this.props.navigation.state.key;
 
+        // Make sure the XML has the required elements: <doc>, <screen>, <body>.
+        const docElement = getFirstTag(doc, 'doc');
+        if (!docElement) {
+          console.error(`No <doc> tag found in the response from ${url}.`);
+          doc = null;
```

Comment:
why do we need to set doc to `null`?
also wondering if a more readable pattern would be to throw an error and let the `catch` handler log the error/set the error state for us

Reply:
just cleaning up the state to not have references to the previous or current document when an error occurs. The previous document can be set in cases where we did a reload of the current screen, or if there was a custom preloading state.